dxr/app.py

```diff
@@ -222,19 +222,21 @@ def _search_html(query, tree, query_text, offset, limit, config):
                                   redirect='false'),
             'top_of_tree': url_for('.browse', tree=tree),
             'tree': tree,
-            'tree_tuples': _tree_tuples(query_text),
+            'tree_tuples': _tree_tuples('.search', q=query_text),
             'www_root': config.www_root}
 
     return render_template('search.html', **template_vars)
 
 
-def _tree_tuples(query_text):
+def _tree_tuples(endpoint, **kwargs):
     """Return a list of rendering info for Switch Tree menu items."""
     return [(f['name'],
-             url_for('.search',
+             url_for(endpoint,
                      tree=f['name'],
-                     q=query_text),
-             f['description'])
+                     **kwargs),
+             f['description'],
+             [(lang, color) for p in plugins_named(f['enabled_plugins'])
+                 for lang, color in p.badge_colors.iteritems()])
```

Comment:
Isn't p.badge_colors going to iterate in arbitrary order? We should be deterministic here, maybe alphabetical.